### PR TITLE
feat: add projects

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: scw/client_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ const (
 	testAccessKey             = "SCW1234567890ABCDEFG"
 	testSecretKey             = "7363616c-6577-6573-6862-6f7579616161" // hint: | xxd -ps -r
 	testDefaultOrganizationID = "6170692e-7363-616c-6577-61792e636f6d" // hint: | xxd -ps -r
+	testDefaultProjectID      = "6170692e-7363-616c-6577-61792e636f6d" // hint: | xxd -ps -r
```
Comment (jerome-quere) [inline scw/client_test.go:21]: Should we change UUID to avoid broken test edge case

File: scw/config.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ const configFileTemplate = `# Scaleway configuration file
 # Your organization ID is the identifier of your account inside Scaleway infrastructure.
 {{ if .DefaultOrganizationID }}default_organization_id: {{ .DefaultOrganizationID }}{{ else }}# default_organization_id: 11111111-1111-1111-1111-111111111111{{ end }}
 
+# Your project ID is the identifier of the project your resources are attached to.
```
Comment (jerome-quere) [inline scw/config.go:42]: Should we add something like this feature is still in Beta ?

File: scw/config_test.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ var (
 	v2ValidInsecure2              = "true"
 	v2ValidSendTelemetry2         = "true"
 	v2ValidDefaultOrganizationID2 = "6d6f7264-6f72-6772-6561-74616761696e" // hint: | xxd -ps -r
+	v2ValidDefaultProjectID2      = "6d6f7264-6f72-6772-6561-74616761696e" // hint: | xxd -ps -r
```
Comment (jerome-quere) [inline scw/config_test.go:24]: Change UUID to avoid overlap

File: scw/config_test.go [hunk 4]
```diff
@@ -30,13 +31,15 @@ var (
 	v2ValidInsecure              = "false"
 	v2ValidSendTelemetry         = "true"
 	v2ValidDefaultOrganizationID = "6170692e-7363-616c-6577-61792e636f6d" // hint: | xxd -ps -r
+	v2ValidDefaultProjectID      = "6170692e-7363-616c-6577-61792e636f6d" // hint: | xxd -ps -r
```
Comment (jerome-quere) [inline scw/config_test.go:34]: Change UUID to avoid overlap

File: scw/config.go [hunk 5]
```diff
@@ -91,7 +95,8 @@ profiles:
 #   myProfile:
 #     access_key: 11111111-1111-1111-1111-111111111111
 #     secret_key: 11111111-1111-1111-1111-111111111111
-#     organization_id: 11111111-1111-1111-1111-111111111111
```
Comment (jerome-quere) [inline scw/config.go:94]: Good catch 👍

File: scw/env_test.go [hunk 6]
```diff
@@ -47,24 +50,28 @@ func TestLoadEnvProfile(t *testing.T) {
 				terraformAccessKeyEnv:    v2ValidAccessKey,
 				terraformSecretKeyEnv:    v2ValidSecretKey,
 				terraformOrganizationEnv: v2ValidDefaultOrganizationID,
-				terraformRegionEnv:       v2ValidDefaultRegion,
+				// Note: projects might break some tf pipelines.
```
Comment (jerome-quere) [inline scw/env_test.go:53]: Why ?

Comment (kindermoumoute) [reply scw/env_test.go]: nvm

File: scw/config_legacy.go [hunk 7]
```diff
@@ -36,6 +36,7 @@ func (v1 *configV1) toV2() *Config {
 	return &Config{
 		Profile: Profile{
 			DefaultOrganizationID: &v1.Organization,
+			DefaultProjectID:      &v1.Organization,
```
[the remark this answers is not in the document]
Comment (kindermoumoute) [reply scw/config_legacy.go:39]:
```suggestion
			DefaultProjectID:      &v1.Organization, // v1 config is not aware of project, so default project is set to organization ID
```
